format/format.go

```diff
@@ -681,6 +681,45 @@ func (f *fumpter) applyPre(c *astutil.Cursor) {
 	case *ast.AssignStmt:
 		// Only remove lines between the assignment token and the first right-hand side expression
 		f.removeLines(f.Line(node.TokPos), f.Line(node.Rhs[0].Pos()))
+
+	// Clothe naked returns
+	case *ast.ReturnStmt:
+		if node.Results == nil { // We have either a naked return, or a function with no return values
+			parents, _ := astutil.PathEnclosingInterval(f.astFile, node.Pos(), node.End())
```

Comment:
`PathEnclosingInterval` is relatively expensive, particularly since it works based on positions. I would avoid it by, for example, keeping a stack of Func types or results lists as we walk the syntax tree.

Reply:
For some more context - PathEnclosingInterval gets more expensive the larger the file is and the more nesting of syntax tree nodes we have. And, if the file is larger, we also have more return statements to run PathEnclosingInterval for. This could lead to formatting of large files getting significantly slower.

Reply:
This one took some experimentation to work right, but I've got it now, I think.  PTAL.